tests/php/Centreon/Domain/Acknowledgement/AcknowledgementServiceTest.php

```diff
@@ -0,0 +1,662 @@
+            $this->accessGroupRepository,
+            $this->monitoringRepository,
+            $this->engineService,
+            $this->entityValidator
+        );
+        $acknowledgementService->filterByContact($this->adminContact);
+
+        $acknowledgement = $acknowledgementService->findOneAcknowledgement(1);
+        $this->assertEquals($acknowledgement, $this->hostAcknowledgement);
+    }
+
+    /**
+     * test findOneAcknowledgement with acl user
+     */
+    public function testFindOneAcknowledgementWithAclUser()
+    {
+        $this->acknowledgementRepository->expects($this->once())
+            ->method('findOneAcknowledgementForNonAdminUser')
+            ->willReturn($this->serviceAcknowledgement);
+
+        $acknowledgementService = new AcknowledgementService(
+            $this->acknowledgementRepository,
+            $this->accessGroupRepository,
+            $this->monitoringRepository,
+            $this->engineService,
+            $this->entityValidator
+        );
+        $acknowledgementService->filterByContact($this->aclContact);
+
+        $acknowledgement = $acknowledgementService->findOneAcknowledgement(2);
+        $this->assertEquals($acknowledgement, $this->serviceAcknowledgement);
+    }
+
+    /**
+     * test findAcknowledgements with admin user
+     */
+    public function testFindAcknowledgementsWithAdminUser()
+    {
+        $this->acknowledgementRepository->expects($this->once())
+            ->method('findAcknowledgementsForAdminUser')
+            ->willReturn([$this->hostAcknowledgement]);
+
+        $acknowledgementService = new AcknowledgementService(
+            $this->acknowledgementRepository,
+            $this->accessGroupRepository,
+            $this->monitoringRepository,
+            $this->engineService,
+            $this->entityValidator
+        );
+        $acknowledgementService->filterByContact($this->adminContact);
+
+        $acknowledgement = $acknowledgementService->findAcknowledgements();
+        $this->assertEquals($acknowledgement, [$this->hostAcknowledgement]);
+    }
+
+    /**
+     * test findAcknowledgements with acl user
+     */
+    public function testFindAcknowledgementsWithAclUser()
+    {
+        $this->acknowledgementRepository->expects($this->once())
+            ->method('findAcknowledgementsForNonAdminUser')
+            ->willReturn([$this->serviceAcknowledgement]);
+
+        $acknowledgementService = new AcknowledgementService(
+            $this->acknowledgementRepository,
+            $this->accessGroupRepository,
+            $this->monitoringRepository,
+            $this->engineService,
+            $this->entityValidator
+        );
+        $acknowledgementService->filterByContact($this->aclContact);
+
+        $acknowledgement = $acknowledgementService->findAcknowledgements();
+        $this->assertEquals($acknowledgement, [$this->serviceAcknowledgement]);
+    }
+
+    /**
+     * test addHostAcknowledgement which is not validated
```

Comment:
do you mean not processed by engine ?

Reply:
Not really
This test that an exception is thrown if the host entity is not well formated (eg: host id is not set)